components/net_traits/request.rs

```diff
@@ -480,6 +482,9 @@ pub struct Request {
     pub https_state: HttpsState,
 }
 
+unsafe impl Send for Request {}
```

Comment:
Why are these unsafe impls necessary?

Reply:
```
error: future cannot be sent between threads safely
   --> /Users/navgattu/Documents/dev-git/servo/components/net/http_loader.rs:716:1
    |
716 | #[async_recursion]
    | ^^^^^^^^^^^^^^^^^^ future created by async block is not `Send`
    |
    = help: within `net_traits::request::Request`, the trait `std::marker::Sync` is not implemented for `std::cell::Cell<()>`
note: future is not `Send` as this value is used across an await
   --> /Users/navgattu/Documents/dev-git/servo/components/net/http_loader.rs:778:40
    |
778 |                 let preflight_result = cors_preflight_fetch(&request, cache, context).await;
    |                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ first, await occurs here, with `&request` maybe used later...
note: `&request` is later dropped here
   --> /Users/navgattu/Documents/dev-git/servo/components/net/http_loader.rs:778:92
    |
778 |                 let preflight_result = cors_preflight_fetch(&request, cache, context).await;
    |                                                             --------                       ^
    |                                                             |
    |                                                             has type `&net_traits::request::Request` which is not `Send`
    = note: required for the cast to the object type `dyn futures::Future<Output = net_traits::response::Response> + std::marker::Send`
    = note: this error originates in the attribute macro `async_recursion` (in Nightly builds, run with -Z macro-backtrace for more info)
```

This is the due to the `async_recursion` macro, which is required since `http_fetch` is a recursive function (among others).

`http_fetch` calls `http_redirect_fetch` which calls `main_fetch` which calls `http_fetch`.

Reply:
I would feel more comfortable removing all uses of Cell/RefCell from those types and wrapping them in Arc and Mutex instead.

Reply:
`Cell`/`RefCell` seems to originate from `ipc-channel`:
https://github.com/servo/ipc-channel/blob/589a62cc92d2a686769ade6ad4fa485a9904a02b/src/platform/inprocess/mod.rs#L140

https://github.com/servo/ipc-channel/blob/589a62cc92d2a686769ade6ad4fa485a9904a02b/src/platform/inprocess/mod.rs#L68

To be clear, the suggestion would be to replace those uses?

Reply:
Simply wrapping the outer type in `Arc<Mutex<...>>` will be enough to allow it to compile: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=2b431b626ded9270ef9c589b198317ef

Reply:
Yes it will. I misunderstood your suggestion above: "removing all uses of Cell/RefCell from those types"

Reply:
We should wrap the ipc-channel types locally.